### PR TITLE
[Merged by Bors] - feat(Algebra/Polynomial/AlgebraMap): add `algEquiv(OfCompEqX|AevalXAddC)_(eq_iff|symm)`

### PR DESCRIPTION
**Pull request by acmepjz**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by acmepjz** [comment]
I don't know if it's `easy`; the added lemmas are marked as `simp`.

File: Mathlib/Algebra/Polynomial/AlgebraMap.lean [hunk 1]
```diff
@@ -294,12 +294,32 @@ def algEquivOfCompEqX (p q : R[X]) (hpq : p.comp q = X) (hqp : q.comp p = X) : R
   refine AlgEquiv.ofAlgHom (aeval p) (aeval q) ?_ ?_ <;>
     exact AlgHom.ext fun _ ↦ by simp [← comp_eq_aeval, comp_assoc, hpq, hqp]
 
+@[simp]
+theorem algEquivOfCompEqX_eq_iff (p q p' q' : R[X])
```
Comment (Ruben-VandeVelde) [inline Mathlib/Algebra/Polynomial/AlgebraMap.lean:298]: Perhaps in all of those lemmas, the non-Prop arguments should be implicit. Will let a maintainer decide

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**